### ISSUE TITLE
How to prevent memory allocation error after changing the network in Google Colab?

### ISSUE THREAD

**Issue by akshattripathi** [googlecolab/colabtools#1339]
I've written a network in keras to find a certain attribute in 3D graph. The graphs represent events in a particle Physics detector and the data specifies the coordinates where there was a hit (which basically corresponds to a certain amount of energy being deposited at a specific coordinate), along with the amount of energy deposited. I first load my data from a '.h5' file on a variable. I then have a function that loops over that data and basically crops the event to reduce the size of the input being fed to the neural network (this process takes about 20 minutes). After that, I have the configuration of my network in Keras and then I train it. It trains fine, but if I change the network slightly to try and see if I can get a higher accuracy, it doesn't even begin to train and gives me a memory allocation error. If I run the step again where I process the data (where I crop it) and load it into a variable, then it trains fine. Has anyone had this problem before and know how to solve it?

- The web browser you are using (Chrome, Firefox, Safari, etc.):
Chrome

- Link (not screenshot!) to a *minimal, public, self-contained* notebook that
  reproduces this issue (click the *Share* button, then *Get Shareable Link*):
The notebook is not self-contained, you'll need the data file:
https://drive.google.com/file/d/10fnrsc5OPJgiWZBJKTi6--aLcjCf9VXy/view?usp=sharing (put this in a folder called 'output_files' that lives in the same directory as the main notebook file given below:)
https://drive.google.com/file/d/1EqyBokNhrtiJ06tKBm_rX9tOFSYarGHd/view?usp=sharing
